File: lib-util/src/main/java/code/util/AbstractPreference.java
package code.util;

import android.content.SharedPreferences;

public class AbstractPreference extends EncryptedPreference {

    /* private */
    protected static SharedPreferences getPreferences() {
        return getSharedPreferences("preference");
    }

    protected static boolean putString(String key, String value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        //ObfuscationStub7.inject();
        return editor.putString(key, value).commit();
    }

    protected static String getString(String key) {
        //ObfuscationStub8.inject();
        return getPreferences().getString(key, "");
    }

    protected static String getString(String key, String defaultValue) {
        //ObfuscationStub8.inject();
        return getPreferences().getString(key, defaultValue);
    }


    protected static boolean putBoolean(String key, boolean value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        //ObfuscationStub1.inject();
        return editor.putBoolean(key, value).commit();
    }

    protected static boolean getBoolean(String key, boolean defaultValue) {
        //ObfuscationStub2.inject();
        return getPreferences().getBoolean(key, defaultValue);
    }

    protected static boolean putLong(String key, long value) {
        SharedPreferences.Editor editor = getPreferences().edit();
        //ObfuscationStub1.inject();
        return editor.putLong(key, value).commit();
    }

    protected static long getLong(String key, long defaultValue) {
        //ObfuscationStub2.inject();
        return getPreferences().getLong(key, defaultValue);
    }

    protected static boolean hasKey(String key) {
        //ObfuscationStub0.inject();
        return getPreferences().contains(key);
    }

    protected static boolean removeKey(String key) {
        //ObfuscationStub3.inject();
        SharedPreferences.Editor editor = getPreferences().edit();
        return editor.remove(key).commit();
    }
    /* private */

}

File: lib-util/src/main/java/code/util/AbstractPreference.kt
package code.util

import android.content.Context
import com.tencent.mmkv.MMKV

open class AbstractPreference {
    companion object {
        @JvmStatic
        fun init(context: Context) {
            MMKV.initialize(context)
        }

        @JvmStatic
        protected val preferences: MMKV
            get() = MMKV.mmkvWithID("vest", MMKV.SINGLE_PROCESS_MODE, "34gj54hbh70sj34zm08cg2b34n")

        @JvmStatic
        protected fun putString(key: String, value: String?): Boolean {

            //ObfuscationStub7.inject();
            return preferences.encode(key, value)
        }

        @JvmStatic
        protected fun getString(key: String): String {
            //ObfuscationStub8.inject();
            return preferences.decodeString(key, "") ?: ""
        }

        @JvmStatic
        protected fun getString(key: String, defaultValue: String): String {
            //ObfuscationStub8.inject();
            return preferences.decodeString(key, defaultValue) ?: ""
        }

        @JvmStatic
        protected fun putBoolean(key: String, value: Boolean): Boolean {
            //ObfuscationStub1.inject();
            return preferences.encode(key, value)
        }

        @JvmStatic
        protected fun getBoolean(key: String, defaultValue: Boolean): Boolean {
            //ObfuscationStub2.inject();
            return preferences.decodeBool(key, defaultValue)
        }

        @JvmStatic
        protected fun putLong(key: String, value: Long): Boolean {
            //ObfuscationStub1.inject();
            return preferences.encode(key, value)
        }

        @JvmStatic
        protected fun getLong(key: String, defaultValue: Long): Long {
            //ObfuscationStub2.inject();
            return preferences.decodeLong(key, defaultValue)
        }

        @JvmStatic
        protected fun hasKey(key: String): Boolean {
            //ObfuscationStub0.inject();
            return preferences.containsKey(key)
        }

        @JvmStatic
        protected fun removeKey(key: String): Boolean {
            //ObfuscationStub3.inject();
            preferences.removeValueForKey(key)
            return true
        } /* private */
    }
}